apps/nova/src/components/TutorProfile/Rating.tsx

```diff
@@ -1,39 +1,67 @@
-import { Card, asFullAssetUrl } from "@litespace/luna";
+import {
+  Button,
+  ButtonSize,
+  ButtonType,
+  Card,
+  asFullAssetUrl,
+  useFormatMessage,
+} from "@litespace/luna";
 import { IRating } from "@litespace/types";
 import React from "react";
 import dayjs from "@/lib/dayjs";
-import { Star } from "react-feather";
+import { Edit, Star, Trash2 } from "react-feather";
+import { useRender } from "@/hooks/render";
+import EditRating from "./EditRating";
 
 const Rating: React.FC<{ rating: IRating.Populated }> = ({ rating }) => {
+  const edit = useRender();
+  const deleteRating = useRender();
+  const intl = useFormatMessage();
+  const showEditDialog = ()=> edit.show();
+  const showDeleteDialog = () => deleteRating.show()
   return (
     <Card>
       <div className="flex flex-col gap-3">
         <div className="flex flex-row gap-3">
           <div className="w-12 h-12">
             <img
-              className="w-full h-full object-cover"
+              className="object-cover w-full h-full"
               src={
                 rating.rater.image
                   ? asFullAssetUrl(rating.rater.image)
                   : "/avatar-1.png"
               }
             />
           </div>
-          <div>
+          <div className="grow">
             <p>{rating.rater.name}</p>
-            <p className="text-base text-foreground-light flex items-center">
+            <p className="flex items-center text-base text-foreground-light">
               <p className="flex flex-row items-center justify-center gap-1">
                 <Star className="w-4 h-4 fill-yellow-500 stroke-yellow-500" />
                 <span>{rating.value}</span>
               </p>
               &nbsp;
-              <span className="border-2 border-dotted border-foreground-light rounded-full" />
+              <span className="border-2 border-dotted rounded-full border-foreground-light" />
               &nbsp;
               <p>{dayjs(rating.createdAt).fromNow()}</p>
             </p>
           </div>
+          <div className="flex gap-3">
+            <Button onClick={showEditDialog} size={ButtonSize.Tiny} type={ButtonType.Primary}>
+              <Edit className="w-4 text-white" />
+            </Button>
+            <Button onClick={showDeleteDialog} size={ButtonSize.Tiny} type={ButtonType.Primary}>
+              <Trash2 className="w-4 text-white" />
+            </Button>
+          </div>
         </div>
         <p>{rating.feedback}</p>
+        <EditRating
+          open={edit.open}
+          close={edit.hide}
+          rate={rating}
+          title={intl("tutor.rate.editmessage")}
+        />
```

Comment:
Use should only be able to edit the rating if he is the **owner**.

Only show the editing button and the editing dialog incase the current user id is the same as the `rating.raterId` 

You can get the current user from the redux state. 

```
cosnt profile = useAppSelector(profileSelectors.value)

profile.value?.user.id === rating.raterid // <--- user can edit this rating. 
```

Reply:
Ok